llvm/lib/Target/RISCV/GISel/RISCVLegalizerInfo.cpp

```diff
@@ -155,6 +155,9 @@ RISCVLegalizerInfo::RISCVLegalizerInfo(const RISCVSubtarget &ST)
 
   getActionDefinitionsBuilder(G_BITREVERSE).maxScalar(0, sXLen).lower();
 
+  getActionDefinitionsBuilder(G_BITCAST).legalIf(
+      all(isScalableVector(0), isScalableVector(1)));
```

Comment:
Don't we need to check the types that are legal for RISC-V?

Reply:
I don't think SDAG does that. I think it leaves it to the consumers of the bitcast to determine wether they can use that type.

Reply:
And the `.getScalarSizeInBits()` of the two vectors must be equal.

Reply:
> And the `.getScalarSizeInBits()` of the two vectors must be equal.

That shouldn't be required. Only the total vector size should matter.

Reply:
For scalable vectors there is no total vector size.

Reply:
> For scalable vectors there is no total vector size.

There is a symbolic representation of the total vector size that TypeSize can represent. As long as the the TypeSizes are both scalable and have the same minimum size, the bitcast is valid.

Reply:
I would assume you need some maximum min elements supported here

Reply:
> It doesn't matter if you claim invalid type combinations are legal, they won't exist in the first place

> I would assume you need some maximum min elements supported here

Are these two statements conflicting? Maximum number of elements would probably come from the maximum number of elements of one of the legal RISC-V types.

Reply:
No. In the legality rules, there's no need to ensure you only claim valid bitcasts are legal. 
Independently of that, I would expect you need to limit this to supported element sizes

Reply:
I've updated this to check to make sure the types are the same size. I've also restricted to claim that its only on legal types since it doesn't really make sense to assume some min/max sized elements here without assuming that the types are legal.